misc/version_updater.py

```diff
@@ -0,0 +1,314 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+"""
+Helper that help changing the version of a tool across the repository.
+"""
+
+import enum
+import re
+import sys
+from argparse import ArgumentParser
+from dataclasses import dataclass
+from fileinput import FileInput
+from typing import Callable, Union
+
+ReplacementPattern = Union[str, Callable[[str], str]]
+
+RawRegexes = list["Replace"]
+
+
+@dataclass
+class Replace:
+    regex: str
+    replaced: ReplacementPattern
+
+    def compile(self, version: str) -> tuple[re.Pattern[str], str]:
+        regex = re.compile(self.regex)
+        replace = (
+            self.replaced.format(version=version)
+            if isinstance(self.replaced, str)
+            else self.replaced(version)
+        )
+
+        return (regex, replace)
+
+
+def hide_patch_version(template: str, separator: str = ".") -> Callable[[str], str]:
+    def _hide_patch_version(version: str) -> str:
+        return template.format(version=separator.join(version.split(".")[:-1]))
+
+    return _hide_patch_version
+
+
+@enum.unique
+class Tool(enum.Enum):
+    Rust = "rust"
+    Python = "python"
+    Poetry = "poetry"
+    Node = "node"
+    WasmPack = "wasm-pack"
+
+
+TOOLS_VERSION: dict[Tool, str] = {
+    Tool.Rust: "1.68.0",
+    Tool.Python: "3.9.5",
+    Tool.Poetry: "1.3.2",
+    Tool.Node: "18.12.0",
+    Tool.WasmPack: "0.11.0",
+}
+
+
+FILES_WITH_VERSION_INFO: dict[str, dict[Tool, RawRegexes]] = {
+    ".github/workflows/ci-python.yml": {
```

Comment:
Use `pathlib::Path` instead of `str`, and use absolute path by doing
```python
BASEDIR = Path(__file__).parent.parent.resolve()
[...]
my_path = BASEDIR / ".github/workflows/ci-python.yml"
```

This is better for typing, avoid potential error if the script is run from arbitrary directory, and allow to use `my_file.read_text()` / `my_file.write_text()` instead of the more cumbersome C style open/read/write

Reply:
by `my_file` did you mean `my_path` ?

the `read_text` seems useful for the check part.
For the update part I'm not so sure, at this steps I'm doing a inplace update where a line may be updated.

Reply:
done